src/common/NetworkResult.hpp

```diff
@@ -2,14 +2,19 @@
 
 #include <QJsonArray>
 #include <QJsonObject>
+#include <QNetworkReply>
 #include <rapidjson/document.h>
 
+#include <optional>
+
 namespace chatterino {
 
 class NetworkResult
 {
 public:
-    NetworkResult(const QByteArray &data, int status);
+    using Error = QNetworkReply::NetworkError;
```

Comment:
We can use qt's phrasing here and call it NetworkError instead, I think that's a bit clearer as to what its purpose is